experience/createRegister.js

```diff
@@ -21,48 +21,46 @@ module.exports = function createRegister (owner) {
       return memo
     }, {})
 
-    var allAvailable = _.every(_.keys(wrappers), function (key) {
-      log.debug('Checking availability of ' + key)
-      if (wrappers[key].isClaimed()) {
-        log.error(key + ' is already claimed')
-        return false
-      } else {
-        log.debug(key + ' is free')
-        return true
-      }
+    var registrations = _.map(_.keys(wrappers), function (key) {
+      log.debug('Registering ' + key)
+      return wrappers[key].register()
     })
 
-    if (allAvailable) {
-      _.each(_.keys(wrappers), function (key) {
-        wrappers[key].claim()
-      })
-
-      onReactReady(function (React) {
+    all(registrations)
+      .then(onReactReady)
+      .then(function (React) {
         cb({
-          render: function (id, fn) {
-            if (released) return
-
-            if (!wrappers[id]) {
-              log(id).warn('Slot not found')
-              return
-            }
-            var success = wrappers[id].render(fn)
-            if (!success) {
-              log(id).error('Failed to render to ' + id)
-            }
-          },
+          render: render,
           release: release
         }, React)
       })
-    }
 
     return release
 
-    function release () {
+    function render (id, fn) {
+      if (released) return
+      var wrapper = wrappers[id]
+
+      if (!wrapper) {
+        boom('Slot "' + id + '" not found, did you forget to register it?')
```

Comment:
Might need more architectural changes, but we should know if it's been registered or not

Reply:
We do know - if the wrapper doesn't exist at this point, it's not been registered. I'll improve the wording.